salt/client/mixins.py

```diff
@@ -323,113 +316,117 @@ def _low(self, fun, low, print_event=True, full_return=False):
             # runner/wheel output during orchestration).
             print_func = None
 
-        namespaced_event = salt.utils.event.NamespacedEvent(
-            event,
+        with salt.utils.event.NamespacedEvent(
+            salt.utils.event.get_event(
+                'master',
+                self.opts['sock_dir'],
+                self.opts['transport'],
+                opts=self.opts,
+                listen=False,
+            ),
             tag,
             print_func=print_func
-        )
-
-        # TODO: test that they exist
-        # TODO: Other things to inject??
-        func_globals = {'__jid__': jid,
-                        '__user__': data['user'],
-                        '__tag__': tag,
-                        # weak ref to avoid the Exception in interpreter
-                        # teardown of event
-                        '__jid_event__': weakref.proxy(namespaced_event),
-                        }
+        ) as namespaced_event:
```

Comment:
I'm not sure how this works as the context manager (and maybe it will become obvious as I read further into this code), but does the `NamespacedEvent` close this `get_event`, or should we nest this like:

```
        with salt.utils.event.get_event(
                'master',
                self.opts['sock_dir'],
                self.opts['transport'],
                opts=self.opts,
                listen=False,
        ) as event:
            with salt.utils.event.NamespacedEvent(event, tag, print_func=print_func) as namespaced_event:
                ...
```

Reply:
NamespacedEvent is also a context manager with this patch and will destroy the event it's namespacing.

Reply:
👍 Destroy all the things! 🔥 

In that case, I am much in favor of this :)